### PR TITLE
Add get-ask, set-ask retrieval commands

### LINKED ISSUE #3843: Expose MarketSetRetrievalAsk on the `lotus-miner` command line
Currently, there is no way to set retrieval parameters from the lotus-miner command line, cause while MarketSetRetrievalAsk is an api to set these parameters, there's no CLI that uses it.

Proposal:
add `lotus-miner retrieval-deals set-ask` and `lotus-miner retrieval-deals get-ask`
`set-ask` should be have similar to the `storage-deals` version -- the main storage price per byte should be required, and the other parameters should be optional.

### PR DESCRIPTION
**Pull request by ingar**
## Summary
Expose `MarketGetRetrievalAsk`, `MarketSetRetrievalAsk` APIs via the CLI.

The `set-ask` command allows you to modify any or all parameters of the current ask.

Resolves #3843

### CODE REVIEW

**Comment by hannahhoward** [comment]
FYI I don't see this conflicting with actors upgrade so it's probably fine to merge

File: cmd/lotus-storage-miner/retrieval-deals.go [hunk 1]
```diff
@@ -154,3 +159,112 @@ var retrievalDealsListCmd = &cli.Command{
...
+	Name:  "set-ask",
+	Usage: "Configure the provider's retrieval ask",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:  "price",
+			Usage: "Set the price of the ask for retrievals (per byte)",
+		},
+		&cli.StringFlag{
+			Name:  "unseal-price",
+			Usage: "Set the price to unseal",
```
Comment (magik6k) [inline cmd/lotus-storage-miner/retrieval-deals.go:173]: We should note the unit (e.g `FIL/GiB`)

File: cmd/lotus-storage-miner/retrieval-deals.go [hunk 2]
```diff
@@ -154,3 +159,112 @@ var retrievalDealsListCmd = &cli.Command{
...
+		},
+		&cli.StringFlag{
+			Name:        "payment-interval",
+			Usage:       "Set the payment interval (in bytes) for retrieval",
+			DefaultText: "1Mb",
```
Comment (magik6k) [inline cmd/lotus-storage-miner/retrieval-deals.go:178]: Nit: units.RAMInBytes will parse this as 2^20, so technically we should be saying `1MiB`

**Review by magik6k — COMMENTED** [review]
Just 2 nits

File: cmd/lotus-storage-miner/retrieval-deals.go [hunk 3]
```diff
@@ -154,3 +159,112 @@ var retrievalDealsListCmd = &cli.Command{
...
+		if err != nil {
+			return err
+		}
+
+		if cctx.IsSet("price") {
+			v, err := types.ParseFIL(cctx.String("price"))
+			if err != nil {
+				return err
+			}
+			ask.PricePerByte = abi.TokenAmount(v)
```
Comment (magik6k) [inline cmd/lotus-storage-miner/retrieval-deals.go:205]: That's per byte, but above we the comment says per GiB.

We _could_ change the comment, but i think it's better ux to actually have it be per GiB, so we just need to divide here by `1 << 30`

**Review by magik6k — COMMENTED** [review]
1 last thing, then LGTM

File: cmd/lotus-storage-miner/retrieval-deals.go (hunk 1 above)
Comment (ingar) [reply cmd/lotus-storage-miner/retrieval-deals.go:173]: I believe this should be on the retrieval price, not on the unseal price.

File: cmd/lotus-storage-miner/retrieval-deals.go (hunk 3 above)
Comment (ingar) [reply cmd/lotus-storage-miner/retrieval-deals.go]: 👍

**Review by magik6k — APPROVED** [review]

**Pull request merged**